azavea/raster-foundry
# Move Rollbar to common project, add to tiler

## Overview

This breaks out the Rollbar code into a separate project so it can be used by both the `app` and `tile` projects. Since we already had broken out the authentication code for the same purpose, this also moves the authentication code and the Rollbar code into a project called `common`. If `common` starts to become a mess later we can reorganize things further.

### Checklist


## Testing Instructions

 * Edit `tile/src/main/scala/routes/MosaicRoutes.scala` to add an error, something like this:
```scala
  def mosaicProject(implicit db: Database): Route =
    pathPrefix(JavaUUID / Segment / "project" / JavaUUID/ IntNumber / IntNumber / IntNumber) { (orgId, userId, projectId, zoom, x, y) =>
      parameter("tag".?) { tag =>
        get {
          complete {
            throw new RuntimeException("Boom!")
            Mosaic(orgId, userId, projectId, zoom, x, y, tag).map { maybeTile =>
              maybeTile.map { tile => pngAsHttpResponse(tile.renderPng()) }
            }
          }
        }
      }
    }
```
 * Fire up the server and navigate to color correction for an ingested scene. No tiles should load.
 * Log into Rollbar and verify that a new error called "Boom!" (or whatever you used) has been reported.

Closes #1010, #1004
